modules/base/src/algorithm/mesh/meshclipping.cpp

```diff
@@ -0,0 +1,609 @@
+                connectedEdges.clear();
+                createdPolygon = true;
+                break;
+            }
+
+            if (i == edges.size() - 1) {
+                throw Exception(
+                    "Found edge, that is not connected to any other edge. This could mean, the "
+                    "clipped mesh was not manifold.");
+            }
+        }
+
+        if (!createdPolygon) {
+            throw Exception("Could not connect edges to manifold polygon");
+        }
+    }
+
+    return polygons;
+}
+
+std::vector<Polygon<Edge3D>> simplifyPolygons(const std::vector<Polygon<Edge3D>>& polygons,
+                                              const float EPSILON) {
+    std::vector<Polygon<Edge3D>> simplifiedPolygons;
+
+    for (const auto& poly : polygons) {
+        if (poly.size() == 0) continue;
+
+        std::vector<Edge3D> simplifiedEdges{poly.get(0)};
+        for (size_t i = 1; i < poly.size(); ++i) {
+            const auto pivotEdge = &simplifiedEdges.back();
+            const auto testEdge = poly.get(i);
+            const auto pivotDir = glm::normalize(pivotEdge->v2 - pivotEdge->v1);
+            const auto testDir = glm::normalize(testEdge.v2 - testEdge.v1);
+            if (glm::abs(glm::dot(pivotDir, testDir) - 1.0f) < EPSILON) {
+                pivotEdge->v2 = testEdge.v2;
+            } else {
+                simplifiedEdges.push_back(testEdge);
+            }
+        }
+
+        Polygon<Edge3D> newPoly(simplifiedEdges.size());
+        for (size_t i = 0; i < simplifiedEdges.size(); ++i) {
+            newPoly.at(i) = simplifiedEdges[i];
+        }
+        simplifiedPolygons.push_back(newPoly);
+    }
+
+    std::vector<Polygon<Edge3D>> notDegenerated;
+    for (const auto& poly : simplifiedPolygons) {
+        if (poly.size() >= 3) notDegenerated.push_back(poly);
+    }
+
+    return notDegenerated;
+}
+
+}  // namespace detail
+
+std::shared_ptr<Mesh> clipMeshAgainstPlane(const Mesh& mesh, const Plane& worldSpacePlane,
+                                           bool capClippedHoles) {
+
+    using namespace detail;
+
+    const float EPSILON = 0.00001f;
+
+    // Perform clipping in data space
+
+    // Transform plane:
+
+    auto worldToData = mesh.getCoordinateTransformer().getWorldToDataMatrix();
+    auto worldToDataNormal = glm::transpose(glm::inverse(worldToData));
+    auto dataSpacePos = vec3(worldToData * vec4(worldSpacePlane.getPoint(), 1.0));
+    auto dataSpaceNormal =
+        glm::normalize(vec3(worldToDataNormal * vec4(worldSpacePlane.getNormal(), 0.0)));
+
+    Plane plane(dataSpacePos, dataSpaceNormal);
+
+    // Extract vertex data from mesh:
+
+    DrawType drawType = DrawType::Triangles;
+    ConnectivityType connectivityType = ConnectivityType::Strip;
+    const std::vector<vec3>* vertexList = nullptr;
+    const std::vector<vec3>* texcoordlist = nullptr;
+    const std::vector<vec4>* colorList = nullptr;
+    const std::vector<unsigned int>* triangleList = nullptr;
+
```

Comment:
At some point we should do this for all meshes.,. Should not be so hard

Reply:
Good point. Though, I would be happy if someone else could do this.